x-pack/plugins/rule_registry/server/search_strategy/search_strategy.ts

```diff
@@ -36,21 +36,37 @@ export const ruleRegistrySearchStrategyProvider = (
   security?: SecurityPluginSetup,
   spaces?: SpacesPluginStart
 ): ISearchStrategy<RuleRegistrySearchRequest, RuleRegistrySearchResponse> => {
-  const es = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
-
+  const internalUserEs = data.search.searchAsInternalUser;
+  const requestUserEs = data.search.getSearchStrategy(ENHANCED_ES_SEARCH_STRATEGY);
   return {
     search: (request, options, deps) => {
+      // SIEM uses RBAC fields in their alerts but also utilizes ES DLS which
+      // is different than every other solution so we need to special case
+      // those requests.
+      let siemRequest = false;
+      if (request.featureIds.length === 1 && request.featureIds[0] === AlertConsumers.SIEM) {
+        siemRequest = true;
+      } else if (request.featureIds.includes(AlertConsumers.SIEM)) {
+        throw new Error(
+          'The ruleRegistryAlertsSearchStrategy search strategy is unable to accommodate requests containing multiple feature IDs and one of those IDs is SIEM.'
+        );
+      }
```

Comment:
nitpick: do you foresee us having to make this distinction in other parts of the code base? If so it might be worth pulling it out

Reply:
Good thought. I don't think so at this point, but we should definitely pull it out if we need to do it again